filament/src/details/Material.cpp

```diff
@@ -71,6 +74,7 @@ struct Material::BuilderDetails {
     size_t mSize = 0;
     MaterialParser* mMaterialParser = nullptr;
     bool mDefaultMaterial = false;
+    std::unordered_map<std::string, std::variant<int32_t, float, bool>> mConstantSpecializations;
```

Comment:
You could use a `CString` here which tends to generate less code.

I guess it was buggy in the previous version because it used `const char*` without a copy.